docs/development/design/telemetry.rst

```diff
@@ -0,0 +1,232 @@
+Collect Data About Builds
+=========================
+
+We may want to take some decisions in the future about deprecations and supported versions.
+Right now we don't have data about the usage of packages and their versions on Read the Docs
+to be able to make a good decision.
+
+.. contents::
+   :local:
+   :depth: 3
+
+Data to be collected
+--------------------
+
+The following data can be collected after installing all dependencies.
+
+Configuration file
+~~~~~~~~~~~~~~~~~~
+
+We are saving the config file in our database,
+but to save some space we are saving it only if it's different than the one from a previous build
+(if it's the same we save a reference to it).
+
+The config file being saved isn't the original one used by the user,
+but the result of merging it with its default values.
+It's saved using a _fake_ ``JSONField``
+(charfield that is transformed to json when creating the model object).
+For these reasons we can't query or download them in bulk without iterating over all objects.
+
+We may also want to have the original config file,
+so we know which settings users are using.
+
+PIP packages
+~~~~~~~~~~~~
+
+We can get a json with all root dependencies with ``pip list``.
+This will allow us to have the name of the packages and their versions used in the build.
+
+.. code-block::
+
+   $ pip list --pre --not-required --local --format json | jq
```

Comment:
About `--not-required`: even if it's more space, aren't we interested in the transitive dependencies as well?

Reply:
Also, TIL about `--format json` 🚀

Reply:
The good thing about listing only the not required ones is that this gives us an idea of what was specified. But perhaps we should remove `--not-required` and try to store what the user specified instead: the reason is that we also want to know who pins their dependencies, and who doesn't.

Reply:
We probably want both. For example, the version of docutils will likely be decided by transitive requirements (eg. our theme) and we don't want full dependency resolution code on our side :)

I think we can store the user-defined requirements, and also the full set.